Sources/VLCLibraryViewController.m

```diff
@@ -766,12 +767,16 @@ - (void)collectionView:(UICollectionView *)collectionView requestToMoveItemAtInd
     BOOL validFileTypeAtFolderPath = ([folderPathItem isKindOfClass:[MLFile class]] || [folderPathItem isKindOfClass:[MLLabel class]]) && [itemPathItem isKindOfClass:[MLFile class]];
 
     if (!validFileTypeAtFolderPath) {
-        VLCAlertView *alert = [[VLCAlertView alloc] initWithTitle:NSLocalizedString(@"FOLDER_INVALID_TYPE_TITLE", nil) message:NSLocalizedString(@"FOLDER_INVALID_TYPE_MESSAGE", nil) cancelButtonTitle:nil otherButtonTitles:@[NSLocalizedString(@"BUTTON_OK", nil)]];
-
-        alert.completion = ^(BOOL cancelled, NSInteger buttonIndex) {
-            [self updateViewContents];
-        };
-        [alert show];
+        [UIAlertController showAlertInViewController:self
+                                               title:NSLocalizedString(@"FOLDER_INVALID_TYPE_TITLE", nil)
+                                             message:NSLocalizedString(@"FOLDER_INVALID_TYPE_MESSAGE", nil)
+                                   cancelButtonTitle:NSLocalizedString(@"BUTTON_OK", nil)
+                                   otherButtonTitles:nil
+                              destructiveButtonTitle:nil
+                                            tapBlock:^(UIAlertController *alertController, NSInteger buttonIndex) {
+                                                [self updateViewContents];
```

Comment:
This is a tricky one, but I think there a memory leak here due to a retain cycle 🤓

The tapBlock closure gets a strong reference to self (VLCLibraryViewController), so when UIAlertController is dismissed it will not be deallocated because of that reference.

We might need to get a weak reference to self instead.

```diff
+__weak typeof(self) weakSelf = self;
[UIAlertController showAlertInViewController:self
                                                title:NSLocalizedString(@"FOLDER_INVALID_TYPE_TITLE", nil)
                                              message:NSLocalizedString(@"FOLDER_INVALID_TYPE_MESSAGE", nil)
                                    cancelButtonTitle:NSLocalizedString(@"BUTTON_OK", nil)
                                    otherButtonTitles:nil
                               destructiveButtonTitle:nil
                                             tapBlock:^(UIAlertController *alertController, NSInteger buttonIndex) {
-                                                [self updateViewContents];
+                                                __strong typeof(weakSelf) strongSelf = weakSelf;
+                                                [strongSelf updateViewContents];
                                                [alertController dismissViewControllerAnimated:YES completion:nil];
                                             }];
```